edit_book.php

```diff
@@ -0,0 +1,44 @@
+<?php
+include 'database.php';
+
+if (isset($_GET['id'])) {
```

Comment:
Refactor the database operations into a repository class for better separation of concerns and maintainability.

```php
class BookRepository {
    private $connection;

    public function __construct($connection) {
        $this->connection = $connection;
    }

    public function updateBook($id, $title, $author, $year) {
        $stmt = $this->connection->prepare("UPDATE books SET title = ?, author = ?, year = ? WHERE id = ?");
        $stmt->bind_param("ssii", $title, $author, $year, $id);
        return $stmt->execute();
    }
}
```

Usage:

```php
$bookRepository = new BookRepository($connection);
if ($bookRepository->updateBook($id, $title, $author, $year)) {
    echo "Book updated successfully!";
    header("Location: list_books.php");
} else {
    echo "Error: " . $connection->error;
}
```